sapi/phpdbg/phpdbg_watch.c

```diff
@@ -138,10 +138,10 @@ const phpdbg_command_t phpdbg_watch_commands[] = {
 bool phpdbg_check_watch_diff(phpdbg_watchtype type, void *oldPtr, void *newPtr) {
 	switch (type) {
 		case WATCH_ON_BUCKET:
-			if (memcmp(&((Bucket *) oldPtr)->h, &((Bucket *) newPtr)->h, sizeof(Bucket) - sizeof(zval) /* key/val comparison */) != 0) {
+			if (memcmp(&((Bucket *) oldPtr)->h, &((Bucket *) newPtr)->h, sizeof(Bucket) - sizeof(zval) /* hash+key comparison */) != 0) {
 				return 2;
```

Comment:
Unrelated thing I just noticed, but the function says it returns `bool` but here it returns `2`.

I am going to blame myself for not catching this when I probably changed the return type, but is the value of `2` important here?

Reply:
See https://heap.space/search?project=PHP-7.0&project=PHP-7.1&full=&defs=&refs=phpdbg_check_watch_diff&path=&hist=&type= (seems to have been irrelevant from the beginning).

Reply:
Yeah indeed it seems irrelevant, maybe it was intended at one point to be used to signal the bucket key has changed, but who knows.